application/backend/app/services/training_configuration_service.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from uuid import UUID
+
+from sqlalchemy.orm import Session
+
+from app.configuration_tools.training_configuration import TrainingConfiguration
+from app.repositories import ModelRevisionRepository, ProjectRepository
+from app.repositories.training_configuration_repo import TrainingConfigurationRepository
+from app.services import ResourceNotFoundError, ResourceType
+from app.supported_models import SupportedModels
+from app.supported_models.default_models import DefaultModels
+
+
+class TrainingConfigurationService:
+    def __init__(self, db_session: Session) -> None:
+        self._db_session = db_session
+        self._training_config_repo = TrainingConfigurationRepository(db_session)
+
+    def get_training_configuration(
+        self,
+        project_id: UUID,
+        model_architecture_id: str | None = None,
+        model_revision_id: UUID | None = None,
+    ) -> TrainingConfiguration:
```

Comment:
I noticed that the endpoint response contains the field `"model_manifest_id"`. I'd suggest to remove it because it's either not applicable (like in my screenshot, where the endpoint is called without specifying any model) or redundant (if the model is already in the request).

<img width="1651" height="110" alt="image" src="https://github.com/user-attachments/assets/f0ccea22-cad7-4bed-ae3a-9a07b5714bf5" />